pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -263,6 +263,15 @@ func (h *machineDefaulterHandler) Handle(ctx context.Context, req admission.Requ
 
 	klog.V(3).Infof("Mutate webhook called for Machine: %s", m.GetName())
 
+	// Enforce that the same clusterID is set for machineSet and machine labels.
+	// Otherwise a discrepancy on the value would leave the machine orphan
+	// and would trigger a new machine creation by the machineSet.
+	// https://bugzilla.redhat.com/show_bug.cgi?id=1857175
+	if m.GetLabels() == nil {
+		m.Labels = make(map[string]string)
+	}
+	m.Labels[MachineClusterIDLabel] = h.clusterID
```

Comment:
Why use the interface accessor on one line, then the struct member on another? Seems odd to me
```suggestion
	if m.Labels == nil {
		m.Labels = make(map[string]string)
	}
	m.Labels[MachineClusterIDLabel] = h.clusterID
```

Reply:
ok, updated.